packages/graph/hash_graph/lib/graph/src/api/web/error.rs

```diff
@@ -0,0 +1,37 @@
+#![allow(
+    clippy::needless_pass_by_value,
+    reason = "This is considered the last-stop for errors from the perspective of the web-api and \
+              taking Owned reports allows us to simply `map_err` and return the appropriate error \
+              response from request handlers"
+)]
+
+use axum::http::StatusCode;
+use error_stack::Report;
+
+use crate::datastore::{BaseIdAlreadyExists, BaseIdDoesNotExist, QueryError};
+
+/// Convert an error report to an HTTP status code depending on values provided.
+/// Assumes the report is created by a _write_ operation in the datastore.
+pub fn modify_report_to_status_code<T>(report: Report<T>) -> StatusCode {
+    if report.downcast_ref::<BaseIdDoesNotExist>().is_some() {
+        return StatusCode::NOT_FOUND;
+    }
+
+    if report.downcast_ref::<BaseIdAlreadyExists>().is_some() {
+        return StatusCode::CONFLICT;
+    }
```

Comment:
Calling `downcast_ref` only returns the latest type. `request` will return *all* types and also search for `Context::provide` implementations. Was `downcast_ref` used on purpose?

Reply:
Thanks for telling me about `contains`, that was my intentional usage. Will move all error handling to the web handlers